rmw_zenoh_cpp/src/detail/attachment_helpers.cpp

```diff
@@ -16,83 +16,93 @@
 
 #include <cstdlib>
 #include <cstring>
-#include <string>
+#include <stdexcept>
+#include <string_view>
+#include <utility>
 
 #include "rmw/types.h"
 
 #include "attachment_helpers.hpp"
 
 namespace rmw_zenoh_cpp
 {
-//==============================================================================
-bool get_gid_from_attachment(
-  const z_attachment_t * const attachment, uint8_t gid[RMW_GID_STORAGE_SIZE])
-{
-  if (!z_check(*attachment)) {
-    return false;
-  }
 
-  z_bytes_t index = z_attachment_get(*attachment, z_bytes_new("source_gid"));
-  if (!z_check(index)) {
-    return false;
-  }
-
-  if (index.len != RMW_GID_STORAGE_SIZE) {
-    return false;
-  }
+attachement_data_t::attachement_data_t(
+  const int64_t _sequence_number,
+  const int64_t _source_timestamp,
+  const uint8_t _source_gid[RMW_GID_STORAGE_SIZE])
+{
+  sequence_number = _sequence_number;
+  source_timestamp = _source_timestamp;
+  memcpy(source_gid, _source_gid, RMW_GID_STORAGE_SIZE);
+}
 
-  memcpy(gid, index.start, index.len);
+attachement_data_t::attachement_data_t(attachement_data_t && data)
+{
+  sequence_number = std::move(data.sequence_number);
+  source_timestamp = std::move(data.source_timestamp);
+  memcpy(source_gid, data.source_gid, RMW_GID_STORAGE_SIZE);
+}
 
-  return true;
+void attachement_data_t::serialize_to_zbytes(z_owned_bytes_t * attachment)
+{
+  ze_owned_serializer_t serializer;
+  ze_serializer_empty(&serializer);
+  ze_serializer_serialize_str(z_loan_mut(serializer), "sequence_number");
+  ze_serializer_serialize_int64(z_loan_mut(serializer), this->sequence_number);
+  ze_serializer_serialize_str(z_loan_mut(serializer), "source_timestamp");
+  ze_serializer_serialize_int64(z_loan_mut(serializer), this->source_timestamp);
+  ze_serializer_serialize_str(z_loan_mut(serializer), "source_gid");
+  ze_serializer_serialize_buf(z_loan_mut(serializer), this->source_gid, RMW_GID_STORAGE_SIZE);
+  ze_serializer_finish(z_move(serializer), attachment);
 }
 
-//==============================================================================
-int64_t get_int64_from_attachment(
-  const z_attachment_t * const attachment, const std::string & name)
+attachement_data_t::attachement_data_t(const z_loaned_bytes_t * attachment)
```

Comment:
```suggestion
attachment_data_t::attachment_data_t(const z_loaned_bytes_t * attachment)
```

Reply:
Fixed in https://github.com/ros2/rmw_zenoh/pull/276/commits/9f78ca46344677a0a36d9c6fd651ee8e620f51b0